tests/integration/helpers.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# flake8: noqa
+
+"""Charm integration test helpers."""
```

Comment:
question: For posterity, can you discuss why are you doing the tests this way?

It seems to me that half of these "helpers" here are not being executed (e.g. `run_sample_workflow`). Am I missing something?

Reply:
`run_sample_workflow` was the only one I saw not being called, I added a call for it to test that the Temporal server was deployed successfully. Was there any other method you had in mind that was not being used?

Reply:
Not really.